plugins/catalog/src/components/createEditLink.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LocationSpec } from '@backstage/catalog-model';
+
+const determineBitBucketBranch = (url: string): string => {
+  const delimiter = '/';
+  const start = 6;
+  const result = url?.split(delimiter).slice(start)[0];
+  return result;
+};
+
+/**
+ * Creates the edit link for components yaml file
+ * @see LocationSpec
+ * @param location The LocationSpec being used to determine entity SCM location
+ * @returns string representing the edit location based on SCM path
+ */
+
+export const createEditLink = (location: LocationSpec): string => {
+  switch (location.type) {
+    case 'github':
+    case 'gitlab':
+      return location.target.replace('/blob/', '/edit/');
+    case 'bitbucket':
+      return location.target.concat(
+        `?mode=edit&spa=0&at=${determineBitBucketBranch(location.target)}`,
+      );
+    case 'url':
+      if (
+        location.target.includes('https://github.com') ||
```

Comment:
See the below code scanning warnings - it should probably be `startsWith` instead of `includes`.

Another note here - the most thorough way of doing this would actually be to inject the app `Config` here. You would do `const url = new URL(location.target)` and iterate through each config entry under [the `integrations` key](https://github.com/spotify/backstage/blob/master/app-config.yaml#L70), and see if the `url.hostname` matches the `host` in the respective section. That will let you learn what the actual type is, and hence what to do with the url

Reply:
Noob question but how would I inject the app-config here? Currently I have it leveraging the url passed in and the git-url-parse information.